sw/device/sca/aes_serial/aes_serial.c

```diff
@@ -0,0 +1,334 @@
+static const uint64_t kDeadline = 500;           //  500us.
+
+// TODO: Remove once there is a CHECK version that does not require test
+// library dependencies.
+#define CHECK(condition, ...)                             \
+  do {                                                    \
+    if (!(condition)) {                                   \
+      /* NOTE: because the condition in this if           \
+         statement can be statically determined,          \
+         only one of the below string constants           \
+         will be included in the final binary.*/          \
+      if (GET_NUM_VARIABLE_ARGS(_, ##__VA_ARGS__) == 0) { \
+        LOG_ERROR("CHECK-fail: " #condition);             \
+      } else {                                            \
+        LOG_ERROR("CHECK-fail: " __VA_ARGS__);            \
+      }                                                   \
+    }                                                     \
+  } while (false)
+
+/**
+ * Convert `from` binary `to` hex.
+ *
+ * @param from input value in binary format.
+ * @param to   output value in hex format.
+ *
+ * @return kSimpleSerialOk on success, kSimpleSerialError otherwise.
+ */
+static simple_serial_result_t hex_value(char from, char *to) {
+  if (from >= '0' && from <= '9') {
+    *to = from - '0';
+  } else if (from >= 'A' && from <= 'F') {
+    *to = from - 'A' + 10;
+  } else if (from >= 'a' && from <= 'f') {
+    *to = from - 'a' + 10;
+  } else {
+    return kSimpleSerialError;
+  }
+  return kSimpleSerialOk;
+}
+
+/**
+ * Convert `num` bytes `from` hex `to` binary format.
+ *
+ * `from` size is expected to by twice as big as `to`.
+ *
+ * @param from input buffer for hex formatted characters.
+ * @param to   output binary buffer.
+ * @param num  number of characters in output buffer.
+ *
+ * @return kSimpleSerialOk on success, kSimpleSerialError otherwise.
+ */
+static simple_serial_result_t a2b_hex(const char *from, char *to, size_t num) {
+  for (int i = 0; i < num; ++i) {
+    char hi, lo;
+    if (hex_value(from[i * 2], &hi) || hex_value(from[i * 2 + 1], &lo)) {
+      return kSimpleSerialError;
+    }
+    to[i] = ((hi & 0xFF) << 4) | (lo & 0xFF);
+  }
+  return kSimpleSerialOk;
+}
+
+/**
+ * Send `data` to UART in hex byte format.
+ */
+static void print_hex_buffer(const char *data, size_t data_len) {
+  static const char b2a_hex_values[16] = "0123456789ABCDEF";
+  for (int i = 0; i < data_len; ++i) {
+    uart_send_char(b2a_hex_values[data[i] >> 4]);
+    uart_send_char(b2a_hex_values[data[i] & 0xF]);
+  }
+}
+
+/**
+ * Send simple serial command response via UART.
+ *
+ * @param cmd_tag  Simple serial command tag.
+ * @param data     Response data. Converted to hex format by this function.
+ * @param data_len data length.
+ */
+static void print_cmd_response(const char cmd_tag, const char *data,
+                               size_t data_len) {
+  uart_send_char(cmd_tag);
+  print_hex_buffer(data, data_len);
```

Comment:
Then, `print_hex_buffer` can be inlined here.

Reply:
Done the inline, even though with uart (for now).